test/integration_jenkins/plugin_dt_test.rb

```diff
@@ -51,47 +61,45 @@ def create_driver(conf = config)
 
   sub_test_case 'tests for #write' do
     test 'Write all records as a JSON array' do
-      nonce = (0...10).map { (65 + rand(26)).chr }.join
+      nonce = (0...10).map { rand(65..90).chr }.join
       puts "Generating random message: #{nonce}"
       d = create_driver
       d.run do
         d.feed('tag', event_time, { 'message' => nonce })
       end
 
       (0...40).each do |i|
-        puts "Getting logs attempt #{i+1}/40"
-
-        log = get_log(nonce)
-
-        break if log != nil
+        puts "Getting logs attempt #{i + 1}/40"
+        break if try_get_log(nonce)
 
         sleep 10
       end
 
-      puts 'got here'
+      raise 'Could not retrieve log after 40 attempts'
     end
   end
 
-  def get_log(nonce)
+  def try_get_log(nonce)
     uri = URI.parse("#{active_gate_url}/api/v2/logs/search?from=now-30m&limit=1000&query=#{nonce}&sort=-timestamp")
     agent = Net::HTTP.new(uri.host, uri.port)
     agent.use_ssl = true
     agent.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
-    req = Net::HTTP::Get.new(uri, { 'User-Agent' => "fluent-plugin-dynatrace-tests v#{Fluent::Plugin::DynatraceOutputConstants.version}" })
+    options = {
+      'User-Agent' => "fluent-plugin-dynatrace-tests v#{Fluent::Plugin::DynatraceOutputConstants.version}"
+    }
+    req = Net::HTTP::Get.new(uri, options)
     req['Authorization'] = "Api-Token #{api_token}"
 
     res = agent.request(req)
 
-    raise "#{res.code} #{res.message}" if not res.is_a?(Net::HTTPSuccess)
+    raise "#{res.code} #{res.message}" unless res.is_a?(Net::HTTPSuccess)
 
     body = JSON.parse(res.body)
-    results = body["results"]
+    results = body['results']
 
-    return nil if results.length == 0
+    return nil if results.length.zero?
 
-    puts results[0]
```

Comment:
I think it would be fine to log the retrieved line once in case someone wants to verify that.